File: CalculatorBackend/src/computer/ComputeDecorator.java
package computer;

import auxiliary.Result;
import core.Main;
import ifc.IfcExpressionProcessor;

public class ComputeDecorator implements IfcExpressionProcessor {
	
	
    @Override
    public String process(String expression) {
        
    	Result result = Main.splitExpression(expression);
        
        if (!result.middle.isEmpty()) {
            String middle = Main.doChunking(result.middle);
            return Main.compute(result.left + Main.compute(middle) + result.right);
        } else {
            return Main.compute(result.left);
        }
    }
}

File: computer/ComputeDecorator.cs
using CalculatorBackend.auxiliary;
using CalculatorBackend.ifc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CalculatorBackend.computer
{
    internal class ComputeDecorator : IfcExpressionProcessor
    { 
        public string Process(string expression)
        {
            Result result = ACalculator.SplitExpression(expression);

            if (!string.IsNullOrEmpty(result.middle))
            {
                string middle = ACalculator.DoChunking(result.middle);
                return ACalculator.Compute(result.left + ACalculator.Compute(middle) + result.right);
            }
            else
            {
                return ACalculator.Compute(result.left);
            }
        }
   }
}